postgres/datadog_checks/postgres/discovery.py

```diff
@@ -0,0 +1,75 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import logging
+from typing import Dict, List
+
+from datadog_checks.base.utils.discovery import Discovery
+from datadog_checks.postgres.connections import MultiDatabaseConnectionPool
+
+AUTODISCOVERY_QUERY: str = """select {columns} from pg_catalog.pg_database where datistemplate = false;"""
+
+
+class PostgresAutodiscovery(Discovery):
+    def __init__(
+        self,
+        global_view_db: str,
+        autodiscovery_config: Dict,
+        log: logging.Logger,
+        conn_pool: MultiDatabaseConnectionPool,
+        default_ttl: int,
+    ) -> None:
+        # parent class asks for includelist to be a dictionary
+        parsed_include = self._parse_includelist(autodiscovery_config.get("include", [".*"]))
+        super(PostgresAutodiscovery, self).__init__(
+            self._get_databases,
+            include=parsed_include,
+            exclude=autodiscovery_config.get("exclude", []),
+            interval=autodiscovery_config.get("refresh", 3600),
+        )
+        self._log = log
+        self._db = global_view_db
+        self._conn_pool = conn_pool
+        self._default_ttl = default_ttl
+        self._max_databases = autodiscovery_config.get("max_databases", 100)
+
+    def _parse_includelist(self, include: List[str]) -> Dict[str, int]:
+        """
+        Convert includelist to a dictionary so the parent class can process it.
+        """
+        ret = {}
+        for item in include:
+            ret[item] = 0
+        return ret
+
+    def get_items(self) -> List[str]:
+        """
+        Get_items() from parent class returns a generator with four objects:
+            > yield pattern, key(item), item, config
+        This function takes the item of interest (dbname) from this four-tuple
+        and returns the full list of database names from the generator.
+        """
+        items = list(super().get_items())
+        items_parsed = [item[1] for item in items]
+        if len(items_parsed) > self._max_databases:
+            items_parsed = items_parsed[: self._max_databases]
```

Comment:
In the long-run, we should sort these databases by something. Maybe discover the top databases by number of issued queries. Out of scope for this pass, but something to start thinking about.

Reply:
will note this down as a future step in the RFC!